tests/nodeos_producer_watermark_test.py

```diff
@@ -0,0 +1,271 @@
+                Utils.cmdError("expected blockNum %s to be produced by any of the valid producers except %s" % (blockNum, lastBlockProducer))
+                Utils.errorExit("Failed because of incorrect block producer order")
+
+            # make sure that the next set of 12 blocks all have the same blockProducer
+            lastBlockProducer=node.getBlockProducerByNum(blockNum)
+            for k in range(0, 12):
+                blockProducer = validBlockProducer(prodsActive, prodsSeen, blockNum, node1)
+                if lastBlockProducer!=blockProducer:
+                    if not reportFirstMissedBlock:
+                        printStr=""
+                        newBlockNum=blockNum-18
+                        for l in range(0,36):
+                            printStr+="%s" % (newBlockNum)
+                            printStr+=":"
+                            newBlockProducer=node.getBlockProducerByNum(newBlockNum)
+                            printStr+="%s" % (newBlockProducer)
+                            printStr+="  "
+                            newBlockNum+=1
+                        Utils.Print("NOTE: expected blockNum %s (started from %s) to be produced by %s, but produded by %s: round=%s, prod slot=%s, prod num=%s - %s" % (blockNum, startingFrom, lastBlockProducer, blockProducer, i, j, k, printStr))
+                    reportFirstMissedBlock=True
+                    break
+                blockNum+=1
+
+    # make sure that we have seen all 21 producers
+    prodsSeenKeys=prodsSeen.keys()
+    if len(prodsSeenKeys) != prodsSize:
+        Utils.cmdError("only saw %s producers of expected %d. At blockNum %s only the following producers were seen: %s" % (len(prodsSeenKeys), prodsSize, blockNum, ",".join(prodsSeenKeys)))
+        Utils.errorExit("Failed because of missing block producers")
+
+    Utils.Debug=temp
+
+
+Print=Utils.Print
+errorExit=Utils.errorExit
+
+from core_symbol import CORE_SYMBOL
+
+args = TestHelper.parse_args({"--prod-count","--dump-error-details","--keep-logs","-v","--leave-running","--clean-run",
+                              "--wallet-port"})
+Utils.Debug=args.v
+totalNodes=4
+cluster=Cluster(walletd=True)
+dumpErrorDetails=args.dump_error_details
+keepLogs=args.keep_logs
+dontKill=args.leave_running
+prodCount=args.prod_count
+killAll=args.clean_run
+walletPort=args.wallet_port
+
+walletMgr=WalletMgr(True, port=walletPort)
+testSuccessful=False
+killEosInstances=not dontKill
+killWallet=not dontKill
+
+WalletdName=Utils.EosWalletName
+ClientName="cleos"
+
+try:
+    TestHelper.printSystemInfo("BEGIN")
+    cluster.setWalletMgr(walletMgr)
+
+    cluster.killall(allInstances=killAll)
+    cluster.cleanup()
+    Print("Stand up cluster")
+    if cluster.launch(prodCount=prodCount, onlyBios=False, pnodes=totalNodes, totalNodes=totalNodes, totalProducers=totalNodes, useBiosBootFile=False, sharedProducers=1) is False:
```

Comment:
I don't want this test to do all the unnecessary work of bootstrapping token and system contracts. The only contract that needs to be deployed for this test is the `eosio.bios` contract so that the `setprods` action can be used to change the producer schedule.